robolectric/src/main/java/org/robolectric/ParameterizedRobolectricTestRunner.java

```diff
@@ -98,6 +102,16 @@ protected void validateConstructor(List<Throwable> errors) {
     }
 
     @Override
+    protected Statement methodBlock(FrameworkMethod method, Config config, AndroidManifest appManifest, SdkEnvironment sdkEnvironment) {
+      configureShadows(sdkEnvironment, config);
+
+      DeepCloner deepCloner = new DeepCloner(sdkEnvironment.getRobolectricClassLoader());
+      parameters = deepCloner.clone(parameters);
```

Comment:
I do not think there is another way.

The problem here is that the parameter objects are created with the system class loader, but the test is later run by the Robolectric class loader. You cannot event pass the parameters to the test class constructor - you will get an error.

Another solution would be calling the parameter generator method again, but with the Robolectric class loader. I think there are two problems with that:
* we should not call that method more than once during a test, this is not what the user would expect
* if the generator is random, or time-dependent, different parameters will show up in the test names and the test

Also please note that not the class-loader is deep-cloned, the parameter objects are.